## Quality assessment of raw sequences
- FastQC
- QUAST

## Alignment tools:
- BLAT
- Bowtie / bowtie2
- Clustal Omega [website](http://www.clustal.org/)
- TopHat2 (low maintenance, replaced by HISAT)
- [HISAT2](https://daehwankimlab.github.io/hisat2/) + [HISAT-genotype](https://github.com/DaehwanKimLab/hisat-genotype)
- BLAST2
- Burrows-Wheeler Aligner (BWA)

## Data Manipulation Tools:
- SRAtoolkit
- BamTools
- SAMtools

## *De novo* assembly tools
- [Velvet](https://github.com/dzerbino/velvet) / [Oases](https://github.com/dzerbino/oases)
- [Ray](https://github.com/sebhtml/ray)
- SOAPdenovo2
- SOAPdenovo
- [TrinityRNAseq](https://github.com/trinityrnaseq/trinityrnaseq/wiki)

  Don't be confused with trinity desktop environment

- SPAdes

## Preprocessing tools
- Cutadapt
- PRINSEQ
- PRINSEQ++
- Scythe
- Sickle
- TagCleaner

## QIIME II

## Reference based assembly tools
- Cufflinks

## Tools for removing / Detecting redundant sequences
- CAP3
- CD-HIT

## General purpose tools:
- BioPerl
- BioPython
- Bio++ Suite
